adr/0002-workflow-integration-points.md

```diff
@@ -0,0 +1,35 @@
+# 2. Provide workflow integration points
+
+Date: 2018-04-27
+
+## Status
+
+Proposed
+
+## Context
+
+A system to handle the creation, processing and orchestration of customized workflows is a required component of Libero's
+currently proposed design.
+
+Users will each have their individual needs and requirements and therefore may decide to use their 
+workflow solution of choice. 
+
+It is therefore important to define clear integration points between a workflow system and the rest of the Libero
+architecture.
+
+## Decision
+
+The workflow solution should provide:
+
+- HTTP endpoint(s) to allow the triggering of workflows
+- A queue (AMQP 0.9.1) to allow the triggering of workflows
+- Publishing of workflow and activity state event messages via queues (AMQP 0.9.1)
+- Adhere to the [Libero event schema](https://github.com/libero/walking-skeleton/blob/master/events/schema.md)
+
+## Consequences
+
+The workflow system in a Libero deployment will be independent and swappable.
+
+Existing systems may need a customized plugin or interface to adhere to the required integration points.
```

Comment:
I wouldn't expect to see anything about Airflow, for example, from the outside as it should be completely wrapped